### PR TITLE
Updated Pantheon recipe to call out `git clone` pre-step

### PR DESCRIPTION
**Pull request by ccharlton**
Added bits that should clear up any confusion about `lando pull` not doing the initial `git clone ...` for users.

Thank you so much for contributing code to lando!

We will get to your PR as soon as we can but in the meantime you might want to
check to make sure you have done the following. **The more boxes you can check
the easier it will be for us to merge in your changes with confidence**

- [ ] My code passes relevant CI status checks
- [ ] My code includes a test if applicable ([see here](https://docs.devwithlando.io/dev/testing.html))
- [ ] My code includes an update to the `CHANGELOG` ([see here](https://github.com/kalabox/lando/tree/master/docs/changelog))
- [x] My code includes documentation updates if relevant.
- [ ] I have pinged [a project committer](https://docs.devwithlando.io/contrib/contributing.html#committers) when I think my code is ready for prime time.

If you are unclear about any of the above please add comments below so we can
improve our process.

### CODE REVIEW

**Comment by ccharlton** [comment]
I forgot to update the line:

> # Initialize a Pantheon specific .lando.yml after getting code (`git clone ...`) from Pantheon

File: docs/tutorials/pantheon.md [hunk 1]
```diff
@@ -55,7 +56,7 @@ Once you've started up your Pantheon site you will need to pull in your database
 
 ### 1. Using `lando pull`
 
-Lando provides a command for Pantheon sites called `lando pull` to get your database and files. **If you do not specify `--code`, `--database` or `--files` then `lando` will use the environment associated with your currently checked out `git branch`.**
+Lando provides a command for Pantheon sites called `lando pull` to get your database and files. **If you do not specify `--code`, `--database` or `--files` then `lando` will use the environment associated with your currently checked out `git branch`.** Please note, `lando pull` **does not** clone repos, it essentially does a `git pull` (merge) for convenience (see #2 above).
```
Comment (serundeputy) [inline docs/tutorials/pantheon.md]: **does not** clone repos, it ~~essentially~~ does a `git pull` (merge)

@achton I think we should eliminate the word *essentially*

Comment (ccharlton) [reply docs/tutorials/pantheon.md]: Yes, it is superfluous.

**Comment by serundeputy** [comment]
thanks @ccharlton merged!

**Comment by pirog** [comment]
@serundeputy @dustinleblanc i think we need to revert this merge and reconsider.

Ideally the document that is linked to at `Read up on how to get a .lando.yml` should contain all the information you need to get started regardless of where you start (eg on a pantheon recipe or elsewhere). That said it should be improved to include the git clone option and probably have its title and emphasis elevated. We definitely dont want to maintain separate specific instructions for each recipe that are common to all recipes. Also, the current three step process that is in there now is going to cause problems for people who `git clone` and then `lando init pantheon`.

So i'm proposing we 

1. Revert this merge, add in @ccharlton's good feedback to the generic getting started doc https://docs.devwithlando.io/started.html
2. Rename this step in our recipe "getting started" sections so its more obvious what its for and that its importance is emphasized.

**Pull request merged**